### PR TITLE
Slack host request

### PR DESCRIPTION
**Pull request by swemail**
To create a workflow where we can interact with request for public host roles via slack.

When requesting public host role this is sent to slack:
Screenshot 2022-10-25 at 10 18 18

When accepted the message is updated to:
Screenshot 2022-10-25 at 10 20 57

When declined the message is updated to:
Screenshot 2022-10-25 at 10 22 24

- [x] Revert seeding of firestore and auth
- [x] Move /slack to own function
- [x] Create staging and prod slack cupcake apps
- [x] Add env variable for what channel to send to (also toggles if sending at all)
- [x] Add secrets
- [x] Add new env variables to workflows

### CODE REVIEW

File: functions/src/api/index.ts [hunk 1]
```diff
@@ -8,12 +8,19 @@ import sentryErrorHandler from './lib/sentry';
...
-import {createRouter} from '../lib/routers';
+import {cerateSlackRouter, createRouter} from '../lib/routers';
+import {slackRouter} from './slack';
+import verifySlackRequest from './lib/verifySlackRequest';
 
 const app = new Koa();
 
 app.on('error', sentryErrorHandler);
 
+const slackIntegrationRouter = cerateSlackRouter();
```
Comment (gewfy) [inline functions/src/api/index.ts:19]: "cereate"

**Review by gewfy — CHANGES_REQUESTED** [review]

File: functions/package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
     "logs": "firebase functions:log",
     "shell": "yarn build && firebase functions:shell",
     "start:remote": "yarn build && firebase emulators:start",
-    "start": "yarn build && firebase emulators:start --project demo-29k-cupcake --import emulatorSeed --export-on-exit",
+    "start": "yarn build && firebase emulators:start --project demo-29k-cupcake --import emulatorSeed",
```
[the remark this answers is not in the document]
Comment (swemail) [reply functions/package.json:16]: No oops here, one way of having valid accounts I thought. But we can script em up instead 😄

File: functions/src/lib/slack.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+import config from './config';
+import {RequestAction} from './constants/requestAction';
+
+const {SLACK_OAUTH_TOKEN, SLACK_BOT_NAME, SLACK_PUBLIC_HOST_REQUESTS_CHANNEL} =
+  config;
+
+const createSlackClient = () => {
+  if (
+    SLACK_BOT_NAME &&
+    SLACK_PUBLIC_HOST_REQUESTS_CHANNEL &&
```
Comment (gewfy) [inline functions/src/lib/slack.ts:11]: I was more thinking that we should use `SLACK_PUBLIC_HOST_REQUESTS_CHANNEL` to determine if we should send the message. That way we can have some messages being sent, and some not. Checking for `SLACK_BOT_NAME`, `SLACK_OAUTH_TOKEN` here makes total sense thought!

**Review by gewfy — CHANGES_REQUESTED** [review]
Nice work! Some minor changes :)

**Review by gewfy — CHANGES_REQUESTED** [review]
Oh one more thing, could we name the slack channels `#app-host-requests` and `#app-host-requests-staging` so they are grouped with the rest of the `#app-` channels?

File: functions/jest.setup.js [hunk 4]
```diff
@@ -9,4 +9,13 @@ process.env = Object.assign(process.env, {
...
+  SLACK_BOT_NAME: 'Some Bot',
+  SLACK_PUBLIC_HOST_REQUESTS_CHANNEL: '#some-channel',
+  SLACK_OAUTH_TOKEN: 'some-slack-oath-token',
+  SLACK_SIGNING_SECRET: 'some-slack-signing-secret',
 });
+
+global.console = {
+  ...console,
+  error: jest.fn(),
+};
```
[the remark this answers is not in the document]
Comment (swemail) [reply functions/jest.setup.js:21]: The SO way I'm afraid, but works

**Review by gewfy — APPROVED** [review]
Mer eller mindre superbt!

**Pull request merged**